src/raftstore/store/peer.rs

```diff
@@ -1012,6 +1020,40 @@ impl Peer {
             || self.pending_merge_state.is_some()
     }
 
+    fn is_merging_strict(&self) -> bool {
+        self.last_proposed_prepare_merge_idx > self.get_store().committed_index()
+            || self.is_merging()
+    }
+
+    // Check if this peer can handle request_snapshot.
+    pub fn ready_to_handle_request_snapshot(&mut self, request_index: u64) -> bool {
+        let reject_reason = if !self.is_leader() {
+            // Only leader can handle request snapshot.
+            "not_leader"
+        } else if self.get_store().applied_index_term() != self.term() {
```

Comment:
Do we also need to check if the applied_index is equal or larger than request index?